src/content/drafts/how-to-create-a-custom-eslint-rule.md

```diff
@@ -0,0 +1,286 @@
+class b4c0n {};
+```
+
+## Who will benefit from this rule?
+
+Now that you have some idea of how your rule will work, you should take some time to think about who would benefit from this rule. Some questions to ask yourself are:
+
+* Is this a rule that would benefit just your company or is it something that other people might want?
+* If this rule **is** specific to your company:
+  * Is it specific to a single project?
+  * Is it something that many different projects could benefit from?
+* If this rule **isn't** specific to your company:
+  * Is it something that benefits *all* JavaScript projects or is it more specific than that? For example, a rule that provides suggestions for TypeScript apps that use GitHub would be something that is beneficial for some projects, whereas a rule that suggests to not use `console.log` would be beneficial for all types of projects. 
+
+Once you have those answers you can then think about...
+
+## Where will the code for the rule live?
+
+There are four main places that new rules are commonly made:
+
+1. In the core [ESLint repository](https://github.com/eslint/eslint/tree/main/lib/rules) 
+    * *For rules that are useful to all JavaScript projects*
+2. In a plugin that already exists 
+    * *For rules that are useful to a specific subset of JavaScript projects*
+3. In a new plugin 
+    * *For rules that are useful to a specific subset of JavaScript projects, but that don't fit into any existing plugin*
+4. Locally in a specific project by utilizing the plugin [eslint-plugin-local-rules](https://www.npmjs.com/package/eslint-plugin-local-rules) 
+    * *For rules that should remain local and / or don't need to be shared*
+
+*Note*: A plugin is a way of sharing rules that relate to a specific type of project. Installing a plugin **does not** turn on all the rules in it. Rather, it just adds more rules that you can choose to apply or not.
+
+I won't go into more detail about what plugins are or how to make them in this article. Instead, I'll focus on showing you how to make a rule in either the core ESLint repository or in a plugin that already exists. If you want to learn more about how to make plugins, though, please check out the [Working with Plugins](https://eslint.org/docs/latest/developer-guide/working-with-plugins) guide on the ESLint website. 
+
+## How does ESLint parse code?
+
+The last thing you need to think about before coding your rule is the topic of parsing code. It's easy for us, as humans, to think, "Oh, I don't want any identifiers with bacon in them." Computers, unfortunately, are not that simple. They need ways to programmatically define what is an identifier vs. a function declaration vs. a literal and so on. 
+
+Fret not, though, as most of this work has already been done for us. ESLint uses something called [ESTree](https://github.com/estree/estree) to parse code. You can think of it as a way of breaking up code into a giant tree that ESLint can then iterate over to check things. 
```

Comment:
Technically, ESTree is the AST format that ESLint uses, so it can't parse code.

ESLint uses a parser (called Espree) that parses JavaScript code into an ESTree-compatble AST format.

So Espree is the parser, ESTree is the AST format, and ESLint is the tool that uses Espree to create the AST. (Though ESLint can also use other parsers like babel/eslint-parser, Espree is the one it ships with.)